packages/muelu/src/MueCentral/MueLu_VerbosityLevel.hpp

```diff
@@ -109,6 +110,40 @@ namespace MueLu {
   //!
   VerbLevel toMueLuVerbLevel(const Teuchos::EVerbosityLevel verbLevel);
 
+  //! global variable that holds the default output rank for new VerboseObjects
+  extern int defaultMueLuOutputRank;
+
+  //! constant that is interpreted as "output on default rank"
+  extern const int OUTPUT_TO_ROOT;
+
+  //! On all MPI ranks that are contained in oldMap, set the default output rank for MueLu to the lowest available rank that is present in newMap.
```

Comment:
It looks like newMap is the result of `Map::removeEmptyProcesses`.  In that case, newMap's communicator is the result of an `MPI_Comm_split` (via `Teuchos::Comm::split`).  Thus, the new communicator must always have a Process 0.  Any MPI communicator `comm` always has process ranks 0, 1, ..., `comm.size()-1`.

Reply:
That's correct. However, we set the MPI rank that does output with respect to `COMM_WORLD`. But I should check that this actually works after more than one rebalancing step.

Reply:
> However, we set the MPI rank that does output with respect to COMM_WORLD

Don't do that -- it's entirely possible for users to create Epetra or Tpetra objects with a subcomm that does not include Process 0 in `MPI_COMM_WORLD`.

Reply:
I agree that we should not be doing that, but that's what currently happens during rebalancing. We have hard coded that rank 0 of `COMM_WORLD` is responsible for all the output. I could add the option to initialize the `defaultMueLuOutputRank` based on the communicator of the matrix on the finest level.

Reply:
@cgcgcg I see.  That's a bigger issue than this PR.  I agree that this PR should be narrow in scope.